### ISSUE TITLE
Removing generated code from source control

### ISSUE THREAD

**Issue by stefanhoelzl** [stefanhoelzl/k64#6]
Hi,

In #5 all generated Code has changed and so I had over thousand files to review.
There is no need to review them, they just make the review process confusing.

It holds also the danger, that the commited genrated code is not up to date.

So my plan is to remove the generated code from this repo.

Advantages:
* browsing through commits is much more clear
* it is clearear what actually changed

Disadvantages:
* you have to checkout the repository and generate the code if you want to see how the code Looks like

@miker1423 I think you use this crate a bit, maybe you have some opinon or suggestion on this?

**Comment by miker1423** [comment]
I was checking other repositories with Peripheral Access Crates, and I saw both, repositories where they have all the generated code and upstream the changes (https://github.com/atsamd-rs/atsamd/pull/90), and others where they have none of the generated code (https://github.com/stm32-rs/stm32-rs). 

Maybe having separeted commits to make the changes to the library and the examples is a good idea, on just one commit all the generated code should be changed and in a subsequent commits the example(s) should be fixed to allign with the new crate generated by svd2rust.

**Comment by stefanhoelzl** [comment]
yes, splitting the commits would solve the review problem, but has some drawbacks:
* each commit should be consistent so that I can checkout a arbitrary commit and than I have everything needed to build the project.
* When Splitting the commits that is not the case anymore since I have to checkout the next commit to get the generated code or even the second next to get the updated examples.
* It is also error prone, since somebody who is new or commits not very often and doesnt remember the workflow can forget it. Or I can checkout the commit where the versions was pushed, but has still the old generated code.
* I also cannot see anymore which implications of a code change has on lets say the examples. Since the examples are updated in another commit.

By integrating the code generation process into the build process, I can ensure detemernistic builds. I only have to run the build step an can be sure to get the right result, without any other steps required.

Google git best practices also suggests, not to commit generated code:
https://deepsource.io/blog/git-best-practices/
https://sethrobertson.github.io/GitBestPractices/#donot

Since I do not see any need for the generated code to be in the repository, I will go on and delete it from version control and integrate the generation process into the build process.

**Comment by miker1423** [comment]
Sounds good, the drawbacks listed are a huge and can potentially break something, so I'm ok with not pushing generated code. 👍

**Comment by miker1423** [comment]
Hello, how can I help you to make a new release of the library?

**Comment by stefanhoelzl** [comment]
I am waiting on rust-embedded/svd2rust#392 to to use `build.rs` for code generation, but there is no activity on this issue.

Do you need a release? Let me know and I will release a new version sometime next week, without code generation or some workaround.

**Comment by miker1423** [comment]
I don't "need" a release, I can continue the work with my own build, but would be nice to have it on crates.io, I understand that the problem is with in svd2rust so I can wait, no problem :).

**Comment by jonas-schievink** [comment]
I don't recommend doing this as it would increase build times for users even more (since now they have to wait for svd2rust and its dependencies to be built, and for svd2rust to process the SVD file and generate code).

**Comment by stefanhoelzl** [comment]
>it would increase build times for users even more.

 thanks for pointing this out, haven't tough about this.
This is really a blocker from using the `build.rs` for code generation.

**Comment by miker1423** [comment]
So the generated code would be kept on the repo?
